### PR TITLE
Fix to not shown Instructor Buttons

### PR DESCRIPTION
**Pull request by nicolasruscher**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [X] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context

Currently instructor buttons are shown to let the instructor assess submissions even if the student did not submit the exam. The instructor now has to set the exam to submitted before being able to da that. This makes the behavior more consistent

### Description

We add additional checks to see if the exam was submitted or not. 

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration -> Exam -> Student Exam -> choose one that was not submitted in time
3. See if the buttons ONLY show up, once you set the exam to submitted. 

### Test Coverage

student-exam-detail-table-row.component.ts: 75%
student-exam-detail.component.ts: 86.75

### CODE REVIEW

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail-table-row/student-exam-detail-table-row.component.html [hunk 1]
```diff
@@ -31,7 +31,7 @@
     </ng-template>
 </td>
 <td>
-    <div *ngIf="course?.isAtLeastInstructor && exercise.type != ExerciseType.QUIZ">
+    <div *ngIf="course?.isAtLeastInstructor && exercise.type !== ExerciseType.QUIZ && studentExam.submitted">
         <div *ngIf="result && exercise.type != ExerciseType.PROGRAMMING">
```
Comment (FrankeLukas) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-detail-table-row/student-exam-detail-table-row.component.html:35]: Can we change the `!=` to `!==` here while we're changing this? (Same for line 48.)

Comment (nicolasruscher) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-detail-table-row/student-exam-detail-table-row.component.html:35]:
```suggestion
        <div *ngIf="result && exercise.type !== ExerciseType.PROGRAMMING">
```

Comment (nicolasruscher) [reply src/main/webapp/app/exam/manage/student-exams/student-exam-detail-table-row/student-exam-detail-table-row.component.html]: Of course, done

**Review by FrankeLukas — APPROVED** [review]
Fix confirmed.

**Pull request merged**